### PR TITLE
Tickets/dm 38239

### PR DESCRIPTION
**Pull request by wvreeven**

### CODE REVIEW

File: python/lsst/ts/pre_commit_conf/pre_commit_hooks.py [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+    config_file_name : `str` | None
+        The name of the configuration file. Set to None if no configuration
+        file exists for this hook.
+    config : `str` | None
+        The configuration of the pre-commit hook as contained by the
+        configuration file. Set to None if no configuration file exists for
+        this hook.
+    pre_commit_config : `str`
+        The YAML string to add to the pre-commit-config.yaml file.
+    optional : `bool`
```
Comment (pkubanek) [inline python/lsst/ts/pre_commit_conf/pre_commit_hooks.py:42]: You can save some lines by providing default values, such as:

   excludable: bool = False
   
Then you can drop executable = False from constructor calls.

**Comment by pkubanek** [comment]
I would like to either increase a bit verbosity of --create operation, or provide --verbose (or similar) option to do so. This is intended behaviour, but the user isn't informed about actions:

```bash
rm .clang-format
generate_pre_commit_conf
```
will silently recreate .clang-format, even when .ts_pre_commit_config.yaml and .pre-commit-config.yaml was already in the project and those aren't touched (of course assuming clang-format: true). Some printout "Writing .clang-format" and similar for others will help a lot.

**Review by pkubanek — APPROVED** [review]
This alone is a great improvement and simplification of the existing code. See comments for some small changes that can be added to make it even better.

**Comment by pkubanek** [comment]
Also it might be worthwhile to investigate how to make CI pass - perhaps exclude --create call, run tests first, and after that --create & install & run standard pre-commit checks for the code.

**Review by tribeiro — APPROVED** [review]
Looks good, I noticed that the list of ignores in flake8 and ruff are different, ruff is missing E133. 

Also, I think in the future we could consider another property of the `PreCommitHookMetadata`, something like `replaces`, which would be a list of hooks a certain hook replaces. For example, we could specify that `ruff` "replaces" flake8, so if it is used, the other one is excluded. 

I know right now ruff is still missing some features that flake8 provides but if one day we can replace them, we could use such feature. We could certainly leave this for the future, but something to keep in mind.

**Pull request merged**